src/main/encryptors/LibsodiumEncryptor.ts

```diff
@@ -28,51 +22,71 @@ export class LibsodiumEncryptor implements Encryptor {
   constructor(private props: LibsodiumEncryptorProps) {}
 
   async encrypt<T, U extends keyof T>(props: EncryptProps<T, U>): Promise<EncryptResult<T, U>> {
-    const { item, fieldsToEncrypt } = props
-    const key = await this.props.keyProvider.generateDataKey()
-    const nonce = randombytes_buf(crypto_secretbox_NONCEBYTES)
```

Comment:
Note the bug: this `nonce` was used multiple times in the loop starting on line 39.